src/util/util.c

```diff
@@ -453,3 +454,29 @@ struct tempfile *make_logfile(char *label) {
     
     return(tf);
 }
+
+// close all file descriptors pointing to a directory
+void fd_cleanup(void) {
+    char *fd_path = (char *)malloc(128);
+    int i;
+
+    singularity_message(DEBUG, "Cleanup file descriptor table\n");
+
+    if ( fd_path == NULL ) {
+        singularity_message(ERROR, "Failed to allocate memory\n");
+        ABORT(255);
+    }
+
+    for ( i = 0; i <= sysconf(_SC_OPEN_MAX); i++ ) {
+        int length;
+        length = snprintf(fd_path, 127, "/proc/self/fd/%d", i);
+        fd_path[length] = '\0';
```

Comment:
There was a question @yhcote had here on the `snprintf()` usage and the null termination...